### PR TITLE
feat: Add LaunchDarkly back and put Search behind feature flag

### PR DESCRIPTION
**Pull request by jcbcapps**
# SC-1499

## Proposed changes
- Adds LaunchDarkly back to the project
- Puts the Search component behind a feature flag
- Adds dark mode styles for Search component
- Adds custom LaunchDarkly context

---

## Reviewer notes
Run the app locally:
- Make sure that you have `export LAUNCHDARKLY_SDK_CLIENT_SIDE_ID=localfile` in your `.envrc` file (or check your .envrc.local)
- While on the homepage, toggle the `searchComponent` flag in `src/stores/launchDarklyLocal.ts` and verify that the expected behavior is shown in the portal (you may need to refresh the page)
- Run `yarn storybook` and check under `Navigation/PageHeader/SearchOn`
- We will also need to test this branch in the dev environment, which will require logging in to LaunchDarkly and updating the feature flag there to see if the proper changes will be reflected in the portal

Note: this PR also addresses the dark mode bug mentioned in [this Shortcut story](https://app.shortcut.com/orbit-truss/story/1474/update-the-search-bar-and-the-end-of-search-results-image-to-support-dark-mode)

---

### As the original developer, I have

- [ ] Met the acceptance criteria, or will meet them in subsequent PRs or stories
  - ... 
- [ ] Created new stories in Storybook if applicable
  - [ ] Checked that all Storybook accessibility checks are passing
- [ ] Created/modified automated unit tests in Jest
  - [ ] Including jest-axe checks when UI changes
- [ ] Created/modified automated E2E tests in Cypress
  - [ ] Including cypress-axe checks when UI changes
  - [ ] Checked that the E2E test build is not failing
- Performed [a11y testing](https://github.com/trussworks/accessibility/blob/master/sample_a11y_testing_process.md):
  - [ ] Checked responsiveness in mobile, tablet, and desktop
  - [ ] Checked keyboard navigability
  - [ ] Tested with [VoiceOver](https://dequeuniversity.com/screenreaders/voiceover-keyboard-shortcuts) in Safari
  - [ ] Checked VO's [rotor menu](https://github.com/trussworks/accessibility/blob/master/README.md#how-to-use-the-rotor-menu) for landmarks, page heading structure and links
  - [ ] Used a browser a11y tool to check for issues (WAVE, axe, ANDI or Accessibility addon tab for Storybook)
- [ ] Requested a design review for user-facing changes
- For any new migrations/schema changes:
  - [ ] Followed guidelines for zero-downtime deploys

### As code reviewer(s), I have

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
  - [ ] Checked that the E2E test build is not failing
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] Considered marking this as accepted even if there are small changes needed
- Performed [a11y testing](https://github.com/trussworks/accessibility/blob/master/sample_a11y_testing_process.md):
  - [ ] Checked responsiveness in mobile, tablet, and desktop
  - [ ] Checked keyboard navigability
  - [ ] Tested with [VoiceOver](https://dequeuniversity.com/screenreaders/voiceover-keyboard-shortcuts) in Safari
  - [ ] Checked VO's [rotor menu](https://github.com/trussworks/accessibility/blob/master/README.md#how-to-use-the-rotor-menu) for landmarks, page heading structure and links
  - [ ] Used a browser a11y tool to check for issues (WAVE, axe, ANDI or Accessibility addon tab for Storybook)

### As a designer reviewer, I have

- [ ] Checked in the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow
- Performed [a11y testing](https://github.com/trussworks/accessibility/blob/master/sample_a11y_testing_process.md):
  - [ ] Checked responsiveness in mobile, tablet, and desktop
  - [ ] Checked keyboard navigability
  - [ ] Tested with [VoiceOver](https://dequeuniversity.com/screenreaders/voiceover-keyboard-shortcuts) in Safari
  - [ ] Checked VO's [rotor menu](https://github.com/trussworks/accessibility/blob/master/README.md#how-to-use-the-rotor-menu) for landmarks, page heading structure and links
  - [ ] Used a browser a11y tool to check for issues (WAVE, axe, ANDI or Accessibility addon tab for Storybook)

### As a test user, I have

- Run through the [Test Script](hhttps://docs.google.com/spreadsheets/d/1eV8UK0aJZ0qrzjnXf5SJ_uRiV7bpsj3yrhRFEkjOvV4/edit?usp=sharing):
  - [ ] On commercial internet in IE11
  - [ ] On commercial internet in Firefox
  - [ ] On commercial internet in Chrome
  - [ ] On commercial internet in Safari
  - [ ] On NIPR in IE11
  - [ ] On NIPR in Firefox
  - [ ] On NIPR in Chrome
  - [ ] On NIPR in Safari
  - [ ] On a mobile device in Firefox
  - [ ] On a mobile device in Chrome
  - [ ] On a mobile device in Safari
- [ ] Added any anomalous behavior to this PR

---

### CODE REVIEW

File: src/components/PageHeader/PageHeader.tsx [hunk 1]
```diff
@@ -19,13 +21,14 @@ const PageHeader = ({
           <Grid col="auto" desktop={{ col: 6 }}>
             {children}
           </Grid>
+
           <Grid col="auto" desktop={{ col: 6 }}>
-            <Search disabled={disableSearch} />
+            {flags && flags.searchComponent ? <Search /> : null}
```
Comment (gidjin) [inline src/components/PageHeader/PageHeader.tsx:26]: Suggestion: I think eventually we do want to remove `disabled` as an option on search. However, removing it at this stage means that the ui will look different than current production with the flag off. Long way to say I think that the Search component should still have it's disabled view and the flag from launch darkly just switches if it's disabled or not. Happy to pair if this comment doesn't make sense. My goal is that when the feature is off the portal looks the same as now, but when the feature is on the search bar is displayed.

**Review by gidjin — CHANGES_REQUESTED** [review]
Looks good, everything functions as expected. But I left one suggestion.

**Review by gidjin — APPROVED** [review]
LGTM! Thanks for the changes. I'm confused as to why the `SearchOn` version of the `PageHeader` component in storybook doesn't seem to include the whole header. But it's minor and I don't think it needs to block this PR.

**Comment by jcbcapps** [comment]
> LGTM! Thanks for the changes. I'm confused as to why the `SearchOn` version of the `PageHeader` component in storybook doesn't seem to include the whole header. But it's minor and I don't think it needs to block this PR.

@gidjin Agreed on the storybook component! I was hoping that @shkeating could help me with it when she's back.

**Comment by gidjin** [comment]
> > LGTM! Thanks for the changes. I'm confused as to why the `SearchOn` version of the `PageHeader` component in storybook doesn't seem to include the whole header. But it's minor and I don't think it needs to block this PR.
> 
> @gidjin Agreed on the storybook component! I was hoping that @shkeating could help me with it when she's back.

My suspicion is that `PageHeader` component isn't included in that version because it's not defined in the default on lines 8-18. I would expect if you put `<PageHeader>` tag around the `<Story />` it would work. But I didn't try it. Unless there is some way to note you want it in the `SearchOn`.

**Review by malworks — APPROVED** [review]
Looks good to me! Worked/looked as expected. I'm not able to test the branch in dark launchly, but I expect another engineer can help you with that piece. :)

**Pull request merged**